lib/ansible/plugins/action/net_template.py

```diff
@@ -93,6 +94,11 @@ def _handle_template(self):
         except IOError:
             return dict(failed=True, msg='unable to load src file')
 
+        searchpath = [working_path, os.path.dirname(source)]
+        if self._task._role is not None:
+            searchpath.insert(1, C.DEFAULT_ROLES_PATH)
```

Comment:
DEFAULT_ROLES_PATH should never be in use by a plugin, it should always use the 'current role path' if it exists or fallback to the playbook path. Another issue is that this is treated as a single directory entry when it could be a list of paths.